src/index.js

```diff
@@ -0,0 +1,146 @@
+const state = {
+    city: 'Honolulu',
+    lat: 21.3099,
+    long: 157.8581,
+    temp: 72,
+};
+
+const convertKtoF = (temp) => {
+    return (temp - 273.15) * (9 / 5) + 32;
+};
+
+const findLatitudeAndLongitude = () => {
+    let location = state.city;
+    axios.get('http://localhost:5000/location',
+    {
+        params: {
+            q: location,
+        },
+    })
+    .then( (response) => {
+        console.log(response);
+        state.lat = response.data[0].lat;
+        state.long = response.data[0].lon;
+        console.log(`response: ${state.city}`);
+        console.log('success in findLatitudeAndLongitude!', state.lat, state.long);
+        
+        getWeather();
+    })
+    .catch( (error) => {
+    console.log(`error: ${state.city}`);
+    console.log('error in findLatitudeAndLongitude!');
+    console.log(error);
+    });
+};
+
+const getWeather = () => {
+    axios
+      .get('http://localhost:5000/weather', {
+        params: {
+          lat: state.lat,
+          lon: state.long,
+        },
+      })
+      .then((response) => {
+        const weather = response.data;
+        state.temp = Math.round(convertKtoF(weather.main.temp));
+        formatTempAndGarden();
+      })
+      .catch((error) => {
+        console.log('Error getting the weather:', error);
+      });
+};
+
+// --------------------------------------------------------------------------------------
+
+const updateSky = () => {
+    let sky = document.getElementById('sky');
+    let skyOption = document.getElementById('sky-select').value;
+    if (skyOption === 'Sunny') {
+        sky.textContent = "☁️☁️✨🌚☁️🌝☁️✨☁️";
+    } else if (skyOption === 'Cloudy') {
+        sky.textContent = "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️";
+    } else if (skyOption === 'Rainy') {
+        sky.textContent = "🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧";
+    } else if (skyOption === 'Snowy') {
+        sky.textContent = "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨";
+    }
+};
+
+const formatTempAndGarden = () => {
+    let temp = state.temp;
+    let color = 'blue';
+    let landscape = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲";
+
+    if (temp >= 80) {
+        color = 'red';
+        landscape = "🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂";
+    } else if (temp >= 70) {
+        color = 'orange';
+        landscape = "🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷";
+    } else if (temp >= 60) {
+        color = 'yellow';
+        landscape = "🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃";
+    } else if (temp >= 50) {
+        color = 'green';
+    } else {
+        color = 'blue';
+        landscape = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲";
+    }
+
+    const temperature = document.getElementById('temp');
+    temperature.textContent = String(state.temp);
+    temperature.className = color;
+    temperature.style.color = color;
+    const garden = document.getElementById('garden');
+    garden.textContent = landscape;
+};
+
+const incrementTemp = () => {
+    state.temp += 1;
+    formatTempAndGarden();
+};
+
+const decrementTemp = () => {
+    state.temp -= 1;
+    formatTempAndGarden();
+};
```

Comment:
Notice that these two methods do essentially the same thing but the difference is -= 1 or += 1 (also in JS you can use ++ or -- which is short hand for doing the same thing).

How could you combine the logic of these two methods into one event handler called handleTempChange()?

What if you pass in the event object as an argument and you check if the event's target has an id `up-arrow` or `down-arrow` and depending on the id then you could ++ or --

```javascript
const handleTempChange = (event) => {
    if (event.target.id === 'increaseTempControl') {
        state.tempDeg++;
    } else {
        state.tempDeg--;
    }
    formatTempAndGarden();
};